velox/duckdb/conversion/DuckWrapper.cpp

```diff
@@ -112,35 +140,50 @@ VectorPtr convert(
   auto vectorType = duckVector.GetVectorType();
   switch (vectorType) {
     case ::duckdb::VectorType::FLAT_VECTOR: {
+      VectorPtr result;
       auto& duckValidity = ::duckdb::FlatVector::Validity(duckVector);
       auto* duckData =
           ::duckdb::FlatVector::GetData<typename OP::DUCK_TYPE>(duckVector);
 
-      // TODO Figure out how to perform a zero-copy conversion.
-      auto result = BaseVector::create(veloxType, size, pool);
-      auto flatResult = result->as<FlatVector<typename OP::VELOX_TYPE>>();
-
       // Some DuckDB vectors have different internal layout and cannot be
       // trivially copied.
       if (duckVector.GetType() == LogicalTypeId::HUGEINT ||
           duckVector.GetType() == LogicalTypeId::TIMESTAMP ||
           duckVector.GetType() == LogicalTypeId::DATE ||
           duckVector.GetType() == LogicalTypeId::VARCHAR) {
+        // TODO Figure out how to perform a zero-copy conversion.
```

Comment:
What's blocking this for date and varchar?

Reply:
+1

Reply:
The purpose of this patch is to replace conversion done through `memcpy` with a zero-copy conversion. Dates are converted to timestamps and zero-copy cannot be used for that. For varchar, we need to convert string buffer whose size is currently unknown.